### PR TITLE
Sam/map

### PR DESCRIPTION
**Pull request by samlawhon**
* Made changes from Lauren's serializer branch that were requested
* Added basic css styling to layout and home page
* Added map

### CODE REVIEW

File: app/assets/stylesheets/navigation.scss [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  .menu {
+    background: $topbar-responsive-bg;
+
+    li:last-of-type {
+      margin-right: 0;
+    }
+
+    a {
+      // color: $space-cadet;
```
Comment (dovakeen118) [inline app/assets/stylesheets/navigation.scss:30]: remove if not being used

File: app/assets/stylesheets/navigation.scss [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      @media screen and (max-width: 39.9375em) {
+        width: 100%;
+        margin: 0.875rem 0;
+      }
+    }
+  }
+
+
+
+
```
Comment (dovakeen118) [inline app/assets/stylesheets/navigation.scss:81]: lots of extra white space here

File: app/controllers/api/v1/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Api::V1::ReviewsController < ApplicationController
+  protect_from_forgery unless: -> { request.format.json? }
```
Comment (dovakeen118) [inline app/controllers/api/v1/reviews_controller.rb:2]: this line is already included in ApplicationController

File: app/javascript/react/components/RestaurantIndex.js [hunk 4]
```diff
@@ -1,43 +1,70 @@
...
-    return(
-      <RestaurantTile
-        name={restaurant.name}
-        id={restaurant.id}
-        key={restaurant.id}
-      />
-    )
-  }) 
+  useEffect(() => {
+  fetch('/api/v1/restaurants?location=boston')
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantIndex.js:10]: fetch should be indented under useEffect

File: app/javascript/react/components/RestaurantIndex.js [hunk 5]
```diff
@@ -1,43 +1,70 @@
...
+              </div>
+          </div>
+        </form>
+        <div className="cell medium-auto medium-cell-block-container">
+          <div className="grid-x grid-padding-x" >
+            <div className="cell medium-6 medium-cell-block-y">
+                {restaurantTileArray}
+            </div>
+            <div className="cell medium-6 medium-cell-block-y">
+              {restaurantData.length !== 0 ? <RestaurantMap restaurantsData={restaurantData} /> : null }
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantIndex.js:61]: Ideally want to handle any logic outside of the return

File: app/javascript/react/components/RestaurantShow.js [hunk 6]
```diff
@@ -1,33 +1,102 @@
...
+    if (response.ok) {
+      return response;
+  } else {
+      const errorMessage = `${response.status} (${response.statusText})`;
+      const error = new Error(errorMessage);
+      throw error;
+    }
+  })
+  .then(response => response.json())
+  .then(body => {console.log(body); setReviewData([...getReviewData, body]);})
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:44]: should remove any `console.log` statements

File: app/javascript/react/components/RestaurantShow.js [hunk 7]
```diff
@@ -1,33 +1,102 @@
...
-    .catch(error => console.error(`Error in fetch: ${error.message}`))
-    }, [])
+import React, { useState, useEffect } from "react";
+import ReviewForm from "./ReviewForm";
+import ReviewTile from './ReviewTile';
+
+const RestaurantShow = props => {
+  const [restaurant, setRestaurant] = useState({});
+  const [getReviewData, setReviewData] = useState([]);
+  const [getNewReview, setNewReview] = useState({
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:8]: state for the review form should live in the review form component

File: app/javascript/react/components/RestaurantShow.js [hunk 8]
```diff
@@ -1,33 +1,102 @@
...
+      const error = new Error(errorMessage);
+      throw error;
+    }
+  })
+  .then(response => response.json())
+  .then(body => {console.log(body); setReviewData([...getReviewData, body]);})
+  .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }
+
+  const handleInputChange = event => {
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:48]: functions required for the review form should live in the review form component

File: app/javascript/react/components/RestaurantShow.js [hunk 9]
```diff
@@ -1,33 +1,102 @@
...
-      </h1>
-  )
+    <div>
+      <h2>{restaurant.name}</h2>
+      <p>Cuisine Type: {restaurant.cuisine}</p>
+      <ReviewForm 
+      id={id} 
+      handleInputChange={handleInputChange} 
+      handleSubmit={handleSubmit}
+      getNewReview={getNewReview}/>
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:94]: prop attributes should be indented, line 91-94

File: app/javascript/react/components/RestaurantShow.js [hunk 10]
```diff
@@ -1,33 +1,102 @@
...
+      <ul>
+        {Object.keys(restaurant).length !== 0 ? reviewTileArray() : null}
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:96]: logic should be handled outside the return

File: app/javascript/react/components/RestaurantShow.js [hunk 11]
```diff
@@ -1,33 +1,102 @@
...
+        const error = new Error(errorMessage);
+        throw(error);
+      }
+    })
+    .then(response => response.json())
+    .then(responseBody => setRestaurant(responseBody))
+  .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }, [])
+
+  const reviewTileArray = () => restaurant.reviews.map((review) => (
```
Comment (dovakeen118) [inline app/javascript/react/components/RestaurantShow.js:75]: this doesn't need to be a function
```
const reviewTileArray = restaurant.reviews.map( ...
```

File: app/javascript/react/components/ReviewForm.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
```
Comment (dovakeen118) [inline app/javascript/react/components/ReviewForm.js:2]: This should be a controlled component with its own state

File: Review-site.drawio [hunk 13]
```diff
@@ -1,41 +1,41 @@
-<mxfile host="app.diagrams.net" modified="2020-10-19T19:21:26.450Z" agent="5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.80 Safari/537.36" etag="8TgZSFYhmG7Q6xmb02zj" version="13.7.9" type="github">
```
Comment (dovakeen118) [inline Review-site.drawio:1]: Are these 3 Draw.io files being used in the application?

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Small changes!

File: Review-site.drawio (hunk 13 above)
Comment (SalimAkram) [reply Review-site.drawio:1]: initially when we created the draw.io file it needed to be saved somewhere I can remove these if its a bummer

File: app/javascript/react/components/RestaurantShow.js (hunk 11 above)
Comment (SalimAkram) [reply app/javascript/react/components/RestaurantShow.js:75]: this was fixed in salims pull request 16 edits

File: app/javascript/react/components/RestaurantShow.js (hunk 10 above)
Comment (SalimAkram) [reply app/javascript/react/components/RestaurantShow.js:96]: this was fixed in salims pull request 16 edits

**Pull request closed without merging**